pkg/server/list_users.go

```diff
@@ -34,5 +37,15 @@ func (s *Server) ListUsers(
 	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
 
 	listUsersQuery := listusers.NewListUsersQuery(s.datastore)
-	return listUsersQuery.ListUsers(ctx, req)
+
+	resp, err := listUsersQuery.ListUsers(ctx, &listusers.ListUsersRequest{
+		ListUsersRequest: req,
+		Depth:            s.resolveNodeLimit,
+	})
+	if err != nil {
+		if errors.Is(err, graph.ErrResolutionDepthExceeded) {
+			return nil, serverErrors.AuthorizationModelResolutionTooComplex
+		}
+	}
```

Comment:
Nit: i've learnt from other APIs that we gain literally nothing by not throwing `AuthorizationModelResolutionTooComplex` directly from the API. This error conversion just makes the code longer.

Reply:
Server errors are reserved for the server RPC layer. Having gRPC and RPC specific errors in internal packages leaks transport level details into internal packages. Packages should be self contained and avoid leaking details.

Reply:
I also like having the server errors separated because it enables us to decouple the server from the lower-level abstractions.